From ebcf87527c32db0b6685c63a450489d2c6727069 Mon Sep 17 00:00:00 2001
From: David Edmundson <kde@davidedmundson.co.uk>
Date: Thu, 29 Oct 2020 11:07:36 +0000
Subject: [PATCH] Autodelete smapsRunnable

The runnable has an early return on error conditions. This means
finished() might never be called.

Now we've removed the accessor and provide the return value in the
signal we can just set autoDelete.

BUG: 428048
---
 processcore/processes_linux_p.cpp       | 2 --
 processcore/read_procsmaps_runnable.cpp | 2 +-
 2 files changed, 1 insertion(+), 3 deletions(-)

diff --git a/processcore/processes_linux_p.cpp b/processcore/processes_linux_p.cpp
index 6882f82..ee5559c 100644
--- a/processcore/processes_linux_p.cpp
+++ b/processcore/processes_linux_p.cpp
@@ -558,8 +558,6 @@ bool ProcessesLocal::updateProcessInfo( long pid, Process *process)
         Q_EMIT processUpdated(pid, { { Process::VmPSS, pss } });
     });
 
-    connect(runnable, &ReadProcSmapsRunnable::finished, runnable, &QObject::deleteLater);
-
     QThreadPool::globalInstance()->start(runnable);
 
     if(!d->readProcStat(dir, process)) success = false;
diff --git a/processcore/read_procsmaps_runnable.cpp b/processcore/read_procsmaps_runnable.cpp
index ecace51..b91466b 100644
--- a/processcore/read_procsmaps_runnable.cpp
+++ b/processcore/read_procsmaps_runnable.cpp
@@ -14,7 +14,7 @@ ReadProcSmapsRunnable::ReadProcSmapsRunnable(const QString &dir)
     : QObject()
     , m_dir(dir)
 {
-    setAutoDelete(false);
+    setAutoDelete(true);
 }
 
 void ReadProcSmapsRunnable::run()
-- 
2.25.1

